c65/monitor.c

```diff
@@ -438,118 +377,173 @@ void cmd_call() {
+        if (addr==start || heat_xs[addr] > xmax) xmax = heat_xs[xaddr=addr];
+        if (n >= nmax) continue;
+
+        brk = breakpoints[addr] & MONITOR_ANY;
+        sym = get_next_symbol_by_value(NULL, addr);
+
+        new = brk & (brk ^ prv);
+
+        if (!sym && !new) continue;
+
+        printf("%.4x", addr);
+        for(/**/; sym; sym = get_next_symbol_by_value(sym, addr))
+            printf("  %s", sym->name);
+        puts("");
+        n++;
+
+        if (new) {   /* did some flags just switch on? */
+            printf("  break");
+            if (new & MONITOR_PC) {
+                for(span=addr; span<0x10000 && breakpoints[span] & MONITOR_PC; span++) /**/;
+                if (--span != addr) printf("  x %.4x.%.4x", addr, span);
+                else printf("  x %.4x", addr);
+            }
+            if (new & MONITOR_READ ) {
+                for(span=addr; span<0x10000 && breakpoints[span] & MONITOR_READ; span++) /**/;
+                if (--span != addr) printf("  r %.4x.%.4x", addr, span);
+                else printf("  r %.4x", addr);
+            }
+            if (new & MONITOR_WRITE) {
+                for(span=addr; span<0x10000 && breakpoints[span] & MONITOR_WRITE; span++) /**/;
+                if (--span != addr) printf("  w %.4x.%.4x", addr, span);
+                else printf("  w %.4x", addr);
+            }
+            puts("");
+            n++;
+        }
+    }
+    if (n >= nmax) puts("...");
+    else if (n == 0) puts("(no labels or breakpoints found)");
+    printf(
+        "hotspots: $%" PRIx64 " read @ %.4x; $%" PRIx64 " write @ %.4x; $%" PRIx64 " execute @ %.4x\n",
+        rmax, raddr, wmax, waddr, xmax, xaddr
+    );
+}
 
 void cmd_set() {
-    const char *vars = "axysp nv bdizc";    /* 0-4 are regs, 6-13 are flags*/
-    uint8_t *regs[] = {&a, &x, &y, &sp}, i, bit;
-    uint16_t v;
-    char *p, *q;
-    p = strtok(NULL, " \t");
-    if (!p) {
+    char *name;
+    int v;
+    name = parse_delim();
+    if (!name) {
         puts("Missing register/flag, expected one of a,x,y,sp,pc or n,v,b,d,i,z,c");
         return;
     }
-    q = strchr(vars, tolower(*p));
-    if (!q) {
+    if (E_OK != parse_int(&v, DEFAULT_REQUIRED) || E_OK != parse_end()) return;
+    if (E_OK != set_reg_or_flag(name, v))
         puts("Unknown register/flag, expected one of a,x,y,sp,pc or n,v,b,d,i,z,c");
-        return;
-    }
-    i = q - vars;
-    p = strtok(NULL, " \t");
-    if (E_OK != _str2val(p, &v, -1, "value")) return;
-
-    if (i < 4) {
-        *(regs[i]) = v;
-    } else if (i==4) { /* PC is a special case */
-        pc = v;
-    } else {
-        bit = 1 << (13-i);
-        if (v) {
-            status |= bit;
-        } else {
-            status &= 255 - bit;
-        }
+}
+
+void cmd_ticks() {
+    /* set or show ticks */
+    int v, err;
+
+    if (E_OK == (err = parse_int(&v, DEFAULT_OPTIONAL))) {
+        ticks = v;
+    } else if (E_MISSING == err) {
+        printf("%llu ticks\n", ticks);
```

Comment:
and here is the other one that is left complaining about the %llu.  Those are the only two warnings I see at the moment.